packages/dev/core/src/FlowGraph/flowGraphExecutionBlock.ts

```diff
@@ -0,0 +1,43 @@
+import type { FlowGraph } from "./flowGraph";
+import { FlowGraphBlock } from "./flowGraphBlock";
+import { FlowGraphConnectionPointDirection, FlowGraphSignalConnectionPoint } from "./flowGraphConnectionPoint";
+
+/**
+ * @experimental
+ * A block that executes some action. Always has an input signal.
+ * Can have one or more output signals.
+ */
+export abstract class FlowGraphExecutionBlock extends FlowGraphBlock {
+    public readonly onStart: FlowGraphSignalConnectionPoint;
+
+    private readonly _signalInputs: FlowGraphSignalConnectionPoint[] = [];
+    private readonly _signalOutputs: FlowGraphSignalConnectionPoint[] = [];
+
+    constructor(graph: FlowGraph) {
+        super(graph);
+
+        this.onStart = this._registerSignalInput("onStart");
+    }
+    /**
+     * Executes the flow graph execution block.
+     */
+    abstract execute(): void;

```

Comment:
Could execute be async? or Promise-based? 
Also - making "execute" abstract also means that all other blocks are in charge of notifying the next blocks. _execute could be an implemented protected function that also deals with async-executed blocks and calls the abstract "execute" that every block needs to implement